client/web/compose/src/components/Public/Record/Modal.vue

```diff
@@ -0,0 +1,132 @@
+<template>
+  <b-modal
+    id="record-modal"
+    v-model="showRecordModal"
+    scrollable
+    body-class="p-0"
+    footer-class="p-0"
+    size="xl"
+  >
+    <template #modal-title>
+      <portal-target
+        name="record-modal-header"
+      />
+    </template>
+
+    <view-record
+      :namespace="namespace"
+      :page="page"
+      :module="module"
+      :record-i-d="recordID"
+      :show-record-modal="showRecordModal"
+    />
+
+    <template #modal-footer>
+      <portal-target
+        name="record-modal-footer"
+        class="w-100 m-0"
+      />
+    </template>
+  </b-modal>
+</template>
+
+<script>
+import record from 'corteza-webapp-compose/src/mixins/record'
+import { compose } from '@cortezaproject/corteza-js'
+import ViewRecord from 'corteza-webapp-compose/src/views/Public/Pages/Records/View'
+import { mapGetters } from 'vuex'
+
+export default {
+  i18nOptions: {
+    namespaces: 'block',
+  },
+
+  name: 'RecordModal',
+
+  components: {
+    ViewRecord,
+  },
+
+  mixins: [
+    record,
+  ],
+
+  props: {
+    namespace: {
+      type: compose.Namespace,
+      required: true,
+    },
+  },
+
+  data () {
+    return {
+      showRecordModal: false,
+      recordID: null,
+      module: null,
+      page: null,
+    }
+  },
+
+  computed: {
+    ...mapGetters({
+      getModuleByID: 'module/getByID',
+      pages: 'page/set',
+    }),
+  },
+
+  watch: {
+    showRecordModal: function(){
+      if(!this.showRecordModal){
+        this.$router.replace({ query: { } })
+      }
+    },
+  },
+
+  created () {
+    this.$root.$on('show-record-modal', this.showModal)
+
+    const { modalRecordID, modalModuleID, modalPageID } = this.$route.query
+    if (modalRecordID) {
```

Comment:
Shouldn't you check for this in the showModal, and just pass this in no matter what?

Reply:
I need this check here, otherwise the modal iinstantiates on the record list page on every page load and I need it to show the modal only if recordID exists (the modal has been opened before). If there is a recordId in the url, I also set showRecordModal to be true, to simulate that a record has been clicked.

Reply:
Exactly, thats why you can do this check, plus moduleID and pageID in the showModal function.